gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -680,6 +680,140 @@ gst_tensor_filter_common_free_property (GstTensorFilterPrivate * priv)
+ * @return Automatically detected framework name (NULL if the framework is not supported or invalid)
+ */
+gchar *
+gst_tensor_filter_framework_auto_detection (const gchar ** model_files,
+    unsigned int num_models)
+{
+  const GstTensorFilterFramework *fw = NULL;
+  gchar *file_extension, *file_extension_2;
+  gchar *fw_priority_str = NULL, *detected_fw = NULL;
+  gchar **fw_priority_str_arr;
+  guint i;
+
+  file_extension = strrchr (model_files[0], '.');
+
+  /* Detect framework based on file extension */
+  if (num_models == 1) {
+    if (g_ascii_strcasecmp (file_extension, ".pb") == 0)
+      fw_priority_str =
+          nnsconf_get_custom_value_string ("filter", "framework_priority_pb");
+    else if (g_ascii_strcasecmp (file_extension, ".tflite") == 0)
+      fw_priority_str =
+          nnsconf_get_custom_value_string ("filter",
+          "framework_priority_tflite");
+    else if (g_ascii_strcasecmp (file_extension, ".graph") == 0)
+      detected_fw = g_strdup ("movidius-ncsdk2");
+    else if (g_ascii_strcasecmp (file_extension, ".circle") == 0)
+      detected_fw = g_strdup ("nnfw");
+    else if (g_ascii_strcasecmp (file_extension,
+            NNSTREAMER_SO_FILE_EXTENSION) == 0)
+      detected_fw = g_strdup ("custom");
+    else if (g_ascii_strcasecmp (file_extension, ".bin") == 0
+        || g_ascii_strcasecmp (file_extension, ".xml") == 0)
+      detected_fw = g_strdup ("openvino");
+    else {
+      g_warning ("%s is not supported or invalid extension \n", file_extension);
+      return detected_fw;
+    }
+  } else if (num_models == 2) {
+    file_extension_2 = strrchr (model_files[1], '.');
+    if ((g_ascii_strcasecmp (file_extension, ".so") == 0
+            && g_ascii_strcasecmp (file_extension_2, ".nb") == 0)
+        || (g_ascii_strcasecmp (file_extension_2, ".so") == 0
+            && g_ascii_strcasecmp (file_extension, ".nb") == 0))
+      detected_fw = g_strdup ("vivante");
+    else {
+      g_warning ("%s is not supported or invalid extension \n", file_extension);
+      return detected_fw;
+    }
+  } else {
+    g_warning ("Invalid the number of models");
+    return detected_fw;
+  }
+
+  /* .pb or .tflite have multiple framework. */
+  if (fw_priority_str != NULL) {
+    fw_priority_str_arr = g_strsplit (fw_priority_str, ",", -1);
+
+    for (i = 0; i < g_strv_length (fw_priority_str_arr); i++) {
+      fw = nnstreamer_filter_find (fw_priority_str_arr[i]);
+
+      if (fw) {
+        detected_fw = g_strdup (fw_priority_str_arr[i]);
+        break;
+      } else {
+        g_debug ("Cannot identify the neural network framework. %s\n",
+            fw_priority_str_arr[i]);
+      }
+    }
+    if (fw == NULL) {
+      g_warning ("Cannot get any neural network framework of %s. \n",
+          file_extension);
+      detected_fw = NULL;
+    }
+
+    g_free (fw_priority_str);
+    g_strfreev (fw_priority_str_arr);
+  }
+
+  g_debug ("Automatically detected framework is %s.\n", detected_fw);
```

Comment:
use macro GST_STR_NULL (detected_fw) instead

Reply:
Or add condition about detected_fw
```
if (detected_fw) {
  detected framework is ...
} else {
 cannot identify the framework ...
}
```

Reply:
Thank you. Updated!